File: ncafe/backend/src/main/java/com/new_cafe/app/backend/entity/Menu.java
package com.new_cafe.app.backend.entity;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Menu {
    private Long id;
    private String korName;
    private String engName;
    private String description;
    private String price;
    private Long categoryId;
    private Boolean isAvailable;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private Category category;
    private List<MenuImage> images;
}

File: ncafe/frontend/types/menu.ts
// types/menu.ts

export type MenuCategory = 'COFFEE' | 'BEVERAGE' | 'DESSERT' | 'BAKERY' | 'OTHER';

export interface MenuCategoryInfo {
  id: string;
  korName: string;
  engName: string;
  icon?: string;
  sortOrder: number;
}

export interface MenuImage {
  id: string;
  url: string;
  isPrimary: boolean;
  sortOrder: number;
}

export interface OptionItem {
  id: string;
  name: string;
  priceDelta: number;
}

export interface MenuOption {
  id: string;
  name: string;
  type: 'radio' | 'checkbox';
  required: boolean;
  items: OptionItem[];
}

export interface Menu {
  id: string;
  korName: string;
  engName: string;
  description: string;
  price: number;
  image: string;
  category?: MenuCategory;
  images?: MenuImage[];
  isAvailable?: boolean;
  isSoldOut?: boolean;
  sortOrder?: number;
  options?: MenuOption[];
  createdAt?: string;
  updatedAt?: string;
}
